pkg/kubectl/cmd/patch.go

```diff
@@ -144,8 +147,11 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 		return fmt.Errorf("unable to parse %q: %v", patch, err)
 	}
 
-	mapper, typer := f.Object()
-	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+	mapper, typer, err := f.UnstructuredObject()
```

Comment:
What does a strategic merge patch `--local` on an unknown resource do?  I expect it to fail (obviously), but I'd like a reasonable message and test to make sure it doesn't result in insanity.